### ISSUE TITLE
sql: error during Eval on parser.RangeCond

### ISSUE THREAD

**Issue by knz** [cockroachdb/cockroach#7411]
Found by #7409:

```
pq: sql/parser/eval.go:1824: unhandled type *parser.RangeCond
    SELECT - CASE WHEN NOT NULL BETWEEN NULL AND NULL THEN 17 ELSE NULL END + + 12; -- /home/kena/go/src/github.com/cockroachdb/sqllogictest/test/random/expr/slt_good_1.test:8777
    SELECT ( + - CASE WHEN NULL BETWEEN NULL AND 93 THEN - CAST ( 90 AS INTEGER ) END ) AS col1; -- /home/kena/go/src/github.com/cockroachdb/sqllogictest/test/random/expr/slt_good_1.test:46258
```

**Comment by knz** [comment]
Some more:
```
    SELECT ( - 4 ) * CASE WHEN NOT NULL NOT BETWEEN NULLIF ( ( + 3 ), + 68 ) AND + 76 + - 6 * + - 82 + - + 1 THEN NULL ELSE + + CAST ( NULL AS INTEGER ) END + + 76; -- /home/kena/go/src/github.com/cockroachdb/sqllogictest/test/random/expr/slt_good_4.test:35789
    SELECT + CASE WHEN NULL NOT BETWEEN 71 AND + 29 THEN + COUNT ( * ) WHEN ( - 13 ) NOT BETWEEN NULL AND ( NULL ) AND CAST ( NULL AS INTEGER ) * + 76 <= ( - 92 + + 93 ) THEN NULL END AS col2; -- /home/kena/go/src/github.com/cockroachdb/sqllogictest/test/random/expr/slt_
    SELECT CASE WHEN NOT NULL NOT BETWEEN - 39 AND - 79 THEN + + 28 END * + MIN ( - + 11 ); -- /home/kena/go/src/github.com/cockroachdb/sqllogictest/test/random/expr/slt_good_4.test:67180
```

**Comment by knz** [comment]
Fixed by #7512  or #7729.
